themes/custom/az_barrio/az_barrio.post_update.php

```diff
@@ -28,3 +28,99 @@ function az_barrio_post_update_split_bootstrap_cdn_version(&$sandbox = NULL) {
     $theme_settings->save();
   }
 }
+
+/**
+ * Updates theme settings for improved schema compliance.
+ *
+ * Updates data types, and reorganizes configuration
+ * to match the updated schema.
+ */
+function az_barrio_post_update_update_theme_settings_schema(&$sandbox = NULL) {
+  $config_factory = \Drupal::configFactory();
+  $theme_settings = $config_factory->getEditable('az_barrio.settings');
+  // Update data types for schema compliance.
+  $type_updates = [
+    // Convert boolean values to proper types.
+    'bootstrap_barrio_enable_color' => FALSE,
+    'bootstrap_barrio_image_fluid' => FALSE,
+    'bootstrap_barrio_navbar_flyout' => FALSE,
+    'bootstrap_barrio_navbar_slide' => FALSE,
+
+    // Convert integer values.
+    'bootstrap_barrio_sidebar_collapse' => 0,
+    'bootstrap_barrio_hide_node_label' => 0,
+    'bootstrap_barrio_sidebar_first_affix' => 0,
+    'bootstrap_barrio_sidebar_second_affix' => 0,
+    'bootstrap_barrio_messages_widget_toast_delay' => 0,
+    'footer_default_logo' => 0,
+
+    // Convert string values.
+    'bootstrap_barrio_float_label' => '',
+    'bootstrap_barrio_navbar_offcanvas' => '',
+  ];
+
+  foreach ($type_updates as $key => $value) {
+    $current_value = $theme_settings->get($key);
+    if ($current_value !== NULL) {
+      $theme_settings->set($key, $value);
```

Comment:
Do we need to try to preserve any existing values here like you are doing in `az_barrio_post_update_convert_footer_default_logo_to_boolean()`?